File: 00369_Combinations.java
import java.util.Scanner;
import java.math.BigInteger;

class Main {
    public static BigInteger fac(int n) {
        BigInteger out = BigInteger.ONE;

        for(int i = 2; i <= n; i++) {
            out = out.multiply(BigInteger.valueOf(i));
        }
        return out;
    }

    public static void main(String args[]) {
        int n, m;
        Scanner s = new Scanner(System.in);

        while(s.hasNext()) {
            n = s.nextInt();
            m = s.nextInt();

            if(n == 0 || m == 0) break;

            BigInteger result = fac(n).divide(fac(n-m).multiply(fac(m)));
            System.out.printf("%d things taken %d at a time is %s exactly.\n", n , m, result.toString());
        }

    }
}

File: Volumn003/00369 - Combinations.cpp
#include <cstdio>

unsigned long long coeff[101][101];
void gen_coeff() {
    coeff[0][0] = coeff[1][0] = coeff[1][1] = 1;

    for(int i = 2; i <= 100; i++) {
        coeff[i][0] = coeff[i][i] = 1;
        for(int j = 0; j <= i; j++) {
            coeff[i][j] = coeff[i-1][j-1] + coeff[i-1][j];
        }
    }
}

int main() {
    gen_coeff();
    int n, k;

    while(scanf("%d %d", &n, &k), n || k) {
        printf("%d things taken %d at a time is %lld exactly.\n", n, k, coeff[n][k]);
    }
}
